### ISSUE TITLE
Cannot Build Godot 4 project

### ISSUE THREAD

**Issue by Kyle-Mendes** [godotengine/godot#65810]
### Godot version

4.0 Alpha 16 & 17 & Godot Beta 4.0

### System information

Mac OS

### Issue description

I'm not really sure what is going wrong, but the export crashes on open when building for all platforms. I can provide a project link privately.  I haven't been able to track down what is happening to make a smaller project to replicate with.

```
arguments
0: /Volumes/harvest.place/harvest.place.app/Contents/MacOS/harvest.place
Current path: /Users/
Godot Engine v4.0.alpha17.official.22a09fef5 - https://godotengine.org
Vulkan API 1.1.216 - Using Vulkan Device #0: AMD - AMD Radeon Pro 5500M
 

================================================================
handle_crash: Program crashed with signal 8
Engine version: Godot Engine v4.0.alpha17.official (22a09fef5d56fc7c37d70118532509076ebd7b12)
Dumping the backtrace. Please include this when reporting the bug to the project developer.
[1] 1   libsystem_platform.dylib            0x00007ff8113c6dfd _sigtramp + 29
[2] 2   ???                                 0x0000000000000040 0x0 + 64
[3] VariantInitializer<Vector<Color> >::init(Variant*)
[4] Object* ClassDB::creator<ENetConnection>()
[5] Object::get_instance_id() const
[6] JSON::get_data() const
[7] VariantInitializer<Vector<Color> >::init(Variant*)
[8] Object* ClassDB::creator<ENetConnection>()
[9] Object::get_instance_id() const
[10] JSON::get_data() const
[11] VariantInitializer<Vector<Color> >::init(Variant*)
[12] Object* ClassDB::creator<ENetConnection>()
[13] MultiplayerAPI::is_server()
[14] Object* ClassDB::creator<ImporterMeshInstance3D>()
[15] ScriptInstanceExtension::_add_property_with_state(void*, void*, void*)
[16] Node::get_tree() const
[17] Node::get_tree() const
[18] Node::is_inside_tree() const
[19] Node::is_inside_tree() const
[20] 20  harvest.place                       0x0000000108e02b9d harvest.place + 4508573
[21] RendererCompositorRD::_create_current()
[22] 22  dyld                                0x000000010f61d52e start + 462
-- END OF BACKTRACE --
================================================================
/Volumes/harvest.place/harvest.place.command: line 15:  7161 Abort trap: 6           "$BASE_PATH/harvest.place.app/Contents/MacOS/harvest.place" "$@"

Saving session...completed.
Deleting expired sessions...       1 completed.

[Process completed]

```

### Steps to reproduce

https://drive.google.com/file/d/1XA08M7_tpP44SyacMNXdOwaqUbGALpk-/view?usp=sharing

**Comment by Kyle-Mendes** [comment]
I was able to debug this down to this line breaking my build:

```
$HTTP.request('http://127.0.0.1:5000/farm/%s' % farm_id, ['Content-Type: application/json'], false, HTTPClient.METHOD_GET)
```

**Comment by Kyle-Mendes** [comment]
I found the offending line!

```
	return arr[randi() % arr.size()]
```

If `arr` is an empty array, this now breaks because I was doing `randi() % 0`.

**Comment by Kyle-Mendes** [comment]
@Chaosus Just wanted to let you know I found the issue :)

**Comment by Kyle-Mendes** [comment]
@akien-mga Just wanted to say that I think there's an improvement to be made in Godot here. This _used_ to work in Exports.  Also, the code works no problem in the editor. It's just at export time that it crashes.  This makes it really hard to debug since there's no feedback in the editor and the debug logs are very opaque.  I only found this after hours of debugging where I removed lines of code until the crash stopped happening.
